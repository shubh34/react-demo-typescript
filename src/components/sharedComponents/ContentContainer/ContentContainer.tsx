import React from 'react';
import { Link } from 'react-router-dom';

interface contentProps {
	blogId: string;
	content: string;
	readMore: boolean;
	className?: string;
}
const ContentContainer: React.FC<contentProps> = props => {
	const { content, readMore, className, blogId } = props;
	return (
		<div className={className}>
			<p>
				{content}
				{readMore && <span>...</span>}
				{readMore && (
					<Link to={`/viewBlog/${blogId}`}>
						<strong>Read More</strong>
					</Link>
				)}
			</p>
		</div>
	);
};

export default ContentContainer;
