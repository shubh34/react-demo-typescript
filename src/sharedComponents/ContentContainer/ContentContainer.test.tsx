import React from 'react';
import { shallow } from 'enzyme';
import ContentContainer from './ContentContainer';


const setup = (setUpProps = {}, container = shallow) => {
	const props = {
	   content :'blog content',
	   readMore : false,
	};
	const wrapper = container(<ContentContainer {...props} />);
	return { wrapper, props };
};

describe('Blog', () => {
	it('renders without crashing', () => {
		const { wrapper } = setup();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render component with read more', () => {
        const { wrapper } = setup({readMore: true});
		expect(wrapper).toMatchSnapshot();
	});
});
