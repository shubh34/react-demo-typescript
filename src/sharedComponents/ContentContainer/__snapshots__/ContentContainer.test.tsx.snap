// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog renders without crashing 1`] = `
<div>
  <p>
    blog content
  </p>
</div>
`;

exports[`Blog should render component with read more 1`] = `
<div>
  <p>
    blog content
  </p>
</div>
`;
